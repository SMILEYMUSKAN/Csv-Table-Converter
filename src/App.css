* {
  box-sizing: border-box;
}

input[type='file'] {
  display: none;
}
.custom-file-upload {
  border: 1px solid dodgerblue;
  display: inline-block;
  padding: 6px 12px;
  cursor: pointer;
  font-style: italic;
  border-radius: 4px;
  color: steelblue;
}

.custom-file-upload:hover {
  background-color: steelblue;
  color: cornsilk;
  transition: all 1s;
}

.main-header {
  text-align: center;
  font-style: italic;
  color: steelblue;
}

.main-parent-div-view {
  max-width: 1000vw;
  display: flex;
  max-height: 1000vh;
  gap: 1em;
}

.left-side-childDiv-screen-view {
  max-width: 900vw;
  min-width: 49%;
  border-style: solid;
  max-height: 900vh;
  min-height: 70vh;
  margin-top: 3em;
  border-color: SteelBlue;
  border-width: 2px;
  text-align: center;
  border-radius: 12px;
}

.left-screen-childDiv-view-main-head {
  color: dodgerblue;
  font-style: italic;
}

.left-chidDiv-section-view {
  margin: 2em;
}
.left-screen-chidDiv-input-view {
  border-radius: 8px;
  padding: 5px;
  background-color: steelblue;
  font-style: italic;
  font-size: 20px;
  cursor: pointer;
}

.leftSide-screen-childDiv-show-csv-view {
  max-width: 500vw;
  min-width: 100%;
  max-height: 800vh;
  min-height: 30vh;
  border: 1px solid dodgerblue;
  margin: auto;
  margin-top: 1em;
  border-radius: 8px;
}

.OnError-left-ChildDiv-csvDataSec-head {
  font-size: 20px;
  font-weight: 5px;
  font-style: italic;
  color: red;
  text-align: center;
}

.OnErrorF-left-ChildDiv-csvDataSec-head {
  text-align: center;
  color: steelblue;
  font-style: italic;
}

.left-childDiv-pre-tag-secView {
  margin: auto;
  box-sizing: border-box;
  margin-top: 1em;
}

.rightside-childDiv-view {
  max-width: 900vw;
  min-width: 49%;
  max-height: 900vh;
  min-height: 70vh;
  border-style: solid;
  margin-top: 3em;
  border-color: SteelBlue;
  border-width: 2px;
  border-radius: 12px;
  box-shadow: 1px 1px 2px black;
}

.right-chidDiv-main-head {
  font-size: 30px;
  font-weight: 5px;
  font-style: italic;
  color: dodgerblue;
  text-align: center;
}

.right-chidDiv-content-head2 {
  text-align: center;
  font-style: italic;
  color: red;
  margin-top: 2em;
}

.right-chidDiv-table-view {
  margin: auto;
  margin-top: 2em;
  padding: 10px;
  border-collapse: collapse;
  border: 1px solid steelblue;
  font-style: italic;
  color: dodgerblue;
}
.right-chidDIv-tbody-style {
  border: 1px solid steelblue;
}

.right-childDiv-td-style {
  border: 1px solid steelblue;
  padding: 10px;
  font-size: 15px;
  color: royalblue;
}
