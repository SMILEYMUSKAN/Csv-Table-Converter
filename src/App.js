import React, { useState, useMemo } from 'react';
import './App.css';
import { isCSVFile, getFileData, csvFileToJsonConverter } from './utils';

export default function App() {
  var [error, setError] = useState(false);
  var [data, setData] = useState(null);
  

  var onFileChange = (event) => {
    var file = event.target.files[0];
    var isValidFile = isCSVFile(file);
    setError(!isValidFile);
    getFileData(file).then((fileData) => setData(fileData))

  };

  var { TableHeader, RemainingTableContent } = useMemo(() => {
    if (!data) {
      return {
        TableHeader: [],
        RemainingTableContent: [],
      };
    }
    return error ? csvFileToJsonConverter('') : csvFileToJsonConverter(data);
  }, [data]);

  console.log(
    ':: HEADER ::',
    TableHeader,
    ':: Remaining Data ::',
    RemainingTableContent
  );

  /*
  useMemo(() => {
    if (!data) {
      return {
        data: [],
      };
    }
    console.log(data);
    var dataSpliter = data.split('\n');
    console.log(dataSpliter, 'dataSpliter');
    var [header, ...remainingData] = dataSpliter;
    console.log(header, remainingData);
    var headerLine = header.split(',');
    console.log(headerLine);
  }, [data]);
  */

  return (
    <main>
      <h1 className="main-header">CSV - Table (Formatter)</h1>
      <div
        style={{
          maxWidth: '1000vw',
          display: 'flex',
          maxheight: '1000vh',
          gap: '1em',
        }}
        className="main-parent-div-view"
      >
        <div className="left-side-childDiv-screen-view">
          <h2 className="left-screen-childDiv-view-main-head">
            Click on "Select File" button to see some magic happen
          </h2>
          <section
            style={{ margin: '2em' }}
            className="left-chidDiv-section-view"
          >
            <label className="custom-file-upload">
              Select File
              <input
                type="file"
                accept=".csv"
                onChange={onFileChange}
                className="left-screen-chidDiv-input-view"
                data-testid="inputFile"
              />
            </label>
            <div className="leftSide-screen-childDiv-show-csv-view">
              {error ? (
                <h2 className="OnError-left-ChildDiv-csvDataSec-head">
                  Please Select Csv File!
                </h2>
              ) : (
                <div>
                  <h2 className="OnErrorF-left-ChildDiv-csvDataSec-head">
                    You can check your csv file here!
                  </h2>
                  <pre className="left-childDiv-pre-tag-secView">{data}</pre>
                </div>
              )}
            </div>
          </section>
        </div>
        <div className="rightside-childDiv-view">
          <h1 className="right-chidDiv-main-head">Table Format</h1>
          {error && (
            <h2 className="right-chidDiv-content-head2">
              Oops! Something Went Wrong
            </h2>
          )}
          {data && (
            <table className="right-chidDiv-table-view">
              <thead>
                <tr>
                  {TableHeader.map((column) => (
                    <th
                      key={column}
                      style={{
                        borderStyle: 'solid',
                        borderColor: `${error ? 'white' : 'steelblue'}`,
                        padding: `${error ? 'none' : '5px'}`,
                        fontSize: '20px',
                        borderWidth: '1px',
                      }}
                    >
                      {column}
                    </th>
                  ))}
                </tr>
              </thead>
              <tbody className="right-chidDIv-tbody-style">
                {RemainingTableContent.map((content, idx) => (
                  <tr key={idx} className="right-chidDIv-tbody-style">
                    {content.map((line) => (
                      <td key={line} className="right-childDiv-td-style">
                        {line}
                      </td>
                    ))}
                  </tr>
                ))}
              </tbody>
            </table>
          )}
        </div>
      </div>
    </main>
  );
}
