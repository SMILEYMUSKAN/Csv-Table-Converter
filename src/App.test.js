import {
  fireEvent,
  render,
  screen,
  waitFor,
  act,
} from "@testing-library/react";
import App from "./App";
// import { isCSVFile } from "./utils";

describe("App File Testing", () => {
  test("Checking htmlElement input is in the document are not", async () => {
    var AppRender = render(<App />);
    var htmlInput = screen.getByTestId("inputFile");
    var Dummy_File_Content = `Name, Job
                            Johnson, UI Dev
                            George, TestEngineer`;
    var Dummy_FileName = "Employee.csv";
    expect(htmlInput).toBeInTheDocument();
    expect(AppRender).toMatchSnapshot();

    const readAsTextMock = jest.fn();
    jest.spyOn(global, "FileReader").mockImplementation(function () {
      const self = this;
      this.readAsText = readAsTextMock.mockImplementation(() => {
        self.onload({ target: { result: Dummy_File_Content } });
      });
    });

    var file = new File([Dummy_File_Content], Dummy_FileName);
    act(() => {
      fireEvent.change(htmlInput, {
        target: { files: [file] },
      });
    });

    await waitFor(() => expect(screen.getByText("UI Dev")).toBeInTheDocument());

    /*
    var Dummy_Error_Content = `Unable to Find File`
    const readAsTextMockError = jest.fn();
    jest.spyOn(global, 'FileReader').mockImplementation(function () {
      const self = this;
      this.readAsText = readAsTextMockError.mockImplementation(() => {
        self.onerror({ target: { result:  Dummy_Error_Content } });
      });
    });
*/
  });

  /*
  test("Testing async test to cover error code on the screen", async () => {
     var InValidFileName = "Index.csv"
     var value = isCSVFile(InValidFileName);
     expect(!value).toEqual(true);
     await waitFor(() => expect( screen.getByText("Please Select Csv File!")).toBeInTheDocument())
     
  })
 */
});
