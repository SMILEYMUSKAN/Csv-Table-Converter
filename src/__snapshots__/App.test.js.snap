// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App File Testing Checking htmlElement input is in the document are not 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <h1
          class="main-header"
        >
          CSV - Table (Formatter)
        </h1>
        <div
          class="main-parent-div-view"
          style="max-width: 1000vw; display: flex; gap: 1em;"
        >
          <div
            class="left-side-childDiv-screen-view"
          >
            <h2
              class="left-screen-childDiv-view-main-head"
            >
              Click on "Select File" button to see some magic happen
            </h2>
            <section
              class="left-chidDiv-section-view"
              style="margin: 2em;"
            >
              <label
                class="custom-file-upload"
              >
                Select File
                <input
                  accept=".csv"
                  class="left-screen-chidDiv-input-view"
                  data-testid="inputFile"
                  type="file"
                />
              </label>
              <div
                class="leftSide-screen-childDiv-show-csv-view"
              >
                <div>
                  <h2
                    class="OnErrorF-left-ChildDiv-csvDataSec-head"
                  >
                    You can check your csv file here!
                  </h2>
                  <pre
                    class="left-childDiv-pre-tag-secView"
                  />
                </div>
              </div>
            </section>
          </div>
          <div
            class="rightside-childDiv-view"
          >
            <h1
              class="right-chidDiv-main-head"
            >
              Table Format
            </h1>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <h1
        class="main-header"
      >
        CSV - Table (Formatter)
      </h1>
      <div
        class="main-parent-div-view"
        style="max-width: 1000vw; display: flex; gap: 1em;"
      >
        <div
          class="left-side-childDiv-screen-view"
        >
          <h2
            class="left-screen-childDiv-view-main-head"
          >
            Click on "Select File" button to see some magic happen
          </h2>
          <section
            class="left-chidDiv-section-view"
            style="margin: 2em;"
          >
            <label
              class="custom-file-upload"
            >
              Select File
              <input
                accept=".csv"
                class="left-screen-chidDiv-input-view"
                data-testid="inputFile"
                type="file"
              />
            </label>
            <div
              class="leftSide-screen-childDiv-show-csv-view"
            >
              <div>
                <h2
                  class="OnErrorF-left-ChildDiv-csvDataSec-head"
                >
                  You can check your csv file here!
                </h2>
                <pre
                  class="left-childDiv-pre-tag-secView"
                />
              </div>
            </div>
          </section>
        </div>
        <div
          class="rightside-childDiv-view"
        >
          <h1
            class="right-chidDiv-main-head"
          >
            Table Format
          </h1>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
